test/params_conversion_test.c

```diff
@@ -0,0 +1,335 @@
+        if (!TEST_true(OSSL_PARAM_set_int64(pc->param, i64))
+            || !TEST_mem_eq(pc->datum, pc->size, pc->ref, pc->size)) {
+            TEST_note("unexpected valid conversion from int64 on line %d",
+                      line);
+            return 0;
+        }
+    }
+
+    if (!pc->valid_u32) {
+        if (!TEST_false(OSSL_PARAM_get_uint32(pc->param, &u32))) {
+            TEST_note("unexpected valid conversion to uint32 on line %d", line);
+            return 0;
+        }
+    } else {
+        if (!TEST_true(OSSL_PARAM_get_uint32(pc->param, &u32))
+            || !TEST_true(u32 == pc->u32)) {
+            TEST_note("unexpected conversion to uint32 on line %d", line);
+            return 0;
+        }
+        memset(pc->datum, 44, pc->size);
+        if (!TEST_true(OSSL_PARAM_set_uint32(pc->param, u32))
+            || !TEST_mem_eq(pc->datum, pc->size, pc->ref, pc->size)) {
+            TEST_note("unexpected valid conversion from uint32 on line %d",
+                      line);
+            return 0;
+        }
+    }
+
+    if (!pc->valid_u64) {
+        if (!TEST_false(OSSL_PARAM_get_uint64(pc->param, &u64))) {
+            TEST_note("unexpected valid conversion to uint64 on line %d", line);
+            return 0;
+        }
+    } else {
+        if (!TEST_true(OSSL_PARAM_get_uint64(pc->param, &u64))
+            || !TEST_true(u64 == pc->u64)) {
+            TEST_note("unexpected conversion to uint64 on line %d", line);
+            return 0;
+        }
+        memset(pc->datum, 44, pc->size);
+        if (!TEST_true(OSSL_PARAM_set_uint64(pc->param, u64))
+            || !TEST_mem_eq(pc->datum, pc->size, pc->ref, pc->size)) {
+            TEST_note("unexpected valid conversion from uint64 on line %d",
+                      line);
+            return 0;
+        }
+    }
+
+    if (!pc->valid_d) {
+        if (!TEST_false(OSSL_PARAM_get_double(pc->param, &d))) {
+            TEST_note("unexpected valid conversion to double on line %d", line);
+            return 0;
+        }
+    } else {
+        if (!TEST_true(OSSL_PARAM_get_double(pc->param, &d))
+            || !TEST_true(d == pc->d)) {
+            TEST_note("unexpected conversion to double on line %d", line);
+            return 0;
+        }
+        memset(pc->datum, 44, pc->size);
+        if (!TEST_true(OSSL_PARAM_set_double(pc->param, d))
+            || !TEST_mem_eq(pc->datum, pc->size, pc->ref, pc->size)) {
+            TEST_note("unexpected valid conversion from double on line %d",
+                      line);
+            return 0;
+        }
+    }
+
+    return 1;
+}
+
+static int run_param_file_tests(int i)
+{
+    STANZA *s;
+    PARAM_CONVERSION pc;
+    const char *testfile = test_get_argument(i);
+    int res = 1;
+
+    if (!TEST_ptr(s = OPENSSL_zalloc(sizeof(*s))))
+        return 0;
+    if (!test_start_file(s, testfile)) {
+        OPENSSL_free(s);
+        return 0;
+    }
+
+    while (!BIO_eof(s->fp)) {
+        test_clearstanza(s);
```

Comment:
You should probably have this at the end of the loop, so the last stanza gets cleared as well.

Reply:
Ah, that makes sense.  I'll fix in the morning.  I was wondering why there were so few leaks.